packages/bindings-factory/lib/Bindings.ts

```diff
@@ -138,13 +191,93 @@ export class Bindings implements RDF.Bindings {
       entries.push([ key, value ]);
     }
 
-    return new Bindings(this.dataFactory, Map(entries));
+    if (this.context) {
+      let mergedContext = this.context;
+      // Only merge if the other has a context
+      if ('context' in other && other.context) {
+        mergedContext = this.mergeContext(mergedContext, other.context);
+      }
+      return new Bindings(this.dataFactory, Map(entries), this.contextMergeHandlers, mergedContext);
+    }
+    const castOther = <Bindings> other;
+    return new Bindings(this.dataFactory, Map(entries), this.contextMergeHandlers, castOther.context);
+  }
+
+  private mergeContext(context: IActionContext, otherContext: IActionContext): IActionContext {
+    // Get Set of all keys present in either of the bindings
+    const keysContext = this.uniqueKeys([ ...context.keys(),
```

Comment:
Can't we just put them in a `Set`? Then we can simplify `uniqueKeys`.

Reply:
The problem is the Set will not consider two key objects with the same name as the equal. If we have
```
const key1 = { name: "key" }; const key2 = { name: "key" }
const set = new Set([key1, key2]);
```
Then set will contain both ```key1``` and ```key2```. 
I could use immutable.js though, if that would be ok here.

Reply:
But in our case, context keys are singletons, so they will always refer to the same entity.
So in this case, the Set should deduplicate them AFAIK.

Reply:
Right, because you use the constant defined in keys.ts? That should work. In the tests, I had them as new objects, which is probably why it didn't work.